airflow/contrib/utils/bluecore_bq_gcs_task_handler.py

```diff
@@ -0,0 +1,182 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from google.cloud import bigquery
+import os
+from airflow.utils.log.gcs_task_handler import GCSTaskHandler
+from airflow import configuration
+from datetime import datetime
+import string
+from airflow.contrib.utils.kubernetes_utils import retryable_check_output
+
+try:
+    import ujson as json
+except ImportError:
+    import json
+
+
+class BQGCSTaskHandler(GCSTaskHandler):
+    """
+    BQGCSTaskHandler is a python log handler that handles and reads
+    task instance logs. It extends airflow FileTaskHandler and GCSTaskHandler
+    uploads to and reads from GCS remote storage. Upon log reading
+    failure, it reads from host machine's local disk. The logs will be displayed in
+    the airflow UI once retrieved from BigQuery.
+    """
+
+    def __init__(self, base_log_folder, gcs_log_folder, filename_template):
+        super(BQGCSTaskHandler, self).__init__(base_log_folder, gcs_log_folder, filename_template)
+
+    def _read(self, ti, try_number):
+        """
+        Read logs of given task instance and try_number from GCS.
+        If failed, read the log from BigQuery.
+
+        :param ti: task instance object
+        :type ti: object
```

Comment:
This is true, but not very descriptive! Type should be `airflow.models.TaskInstance`